### PR TITLE
Fixes #7313: Invalid setpoint yaw after takeoff

### LINKED ISSUE #7313: Invalid setpoint yaw after takeoff
https://github.com/PX4/Firmware/blob/master/src/modules/navigator/mission_block.cpp#L528

I believe sp.yaw_valid should be set correctly in the MissionBlock::mission_item_to_position_setpoint according to (mission) item->yaw.
When yaw_valid is already true (in my case it was set when taking off using MAV_CMD_NAV_TAKEOFF) and mission_item_to_position_setpoint is called with mission item containing yaw=NAN (is happens whan take-off is finished: https://github.com/PX4/Firmware/blob/master/src/modules/navigator/takeoff.cpp#L95), the yaw remains set to NAN and the yaw_valid remains true. When I switch to offboard (without valid yaw set), NAN value from yaw propagates thru the control and causes invalidating attitude setpoint (most values are set to NAN). This leads to instant crash.

### PR DESCRIPTION
**Pull request by Seanmatthews**

### CODE REVIEW

**Comment by Seanmatthews** [comment]
Unborked commit from #7400

File: src/modules/navigator/mission_block.cpp [hunk 1]
```diff
@@ -519,6 +519,7 @@ MissionBlock::mission_item_to_position_setpoint(const struct mission_item_s *ite
 	sp->lon = item->lon;
 	sp->alt = item->altitude_is_relative ? item->altitude + _navigator->get_home_position()->alt : item->altitude;
 	sp->yaw = item->yaw;
+	sp->yaw_valid = !PX4_ISFINITE(item->yaw);
```
Comment (mhkabir) [inline src/modules/navigator/mission_block.cpp]: Incorrect. This should be `PX4_ISFINITE`, not `!PX4_ISFINITE`.

**Comment by mhkabir** [comment]
@MaEtUgR @Stifael please review and merge.

**Comment by mhkabir** [comment]
Thanks!

**Pull request merged**